src/components/structures/RoomView.js

```diff
@@ -681,7 +687,10 @@ module.exports = React.createClass({
             return ret;
         }
 
-        for (var i = this.state.room.timeline.length-1; i >= 0 && count < this.state.messageCap; --i) {
+
+        var prevEvent = null; // the last event we showed
+        var startIdx = Math.max(0, this.state.room.timeline.length-this.state.messageCap);
```

Comment:
nitpick - lack of whitespace around the binary operator does my 'ead in.  ignoring syntax highlighting, it looks to me like a field name called length-this :)